build.gradle

```
@@ -63,7 +63,7 @@ validateNebulaPom.enabled = false
 
 buildscript {
     ext {
-        opensearch_version = System.getProperty("opensearch.version", "3.0.0-SNAPSHOT")
+        opensearch_version = System.getProperty("opensearch.version", "2.15.0-SNAPSHOT")
```

We should be using latest release there:

```suggestion
        opensearch_version = System.getProperty("opensearch.version", "2.14.0")
```

Could you please also update `.github/workflows/CI.yml` to use `2.15.0-SNAPSHOT` (instead of `2.1.0-SNAPSHOT`), thank you:

```
./gradlew updateVersion -DnewVersion=2.15.0-SNAPSHOT
```

---

i had understood that we should always depend on the snapshot version here, based on [this documentation](https://github.com/opensearch-project/opensearch-plugins/blob/main/BUILDING.md#developing-new-plugins-for-opensearch).

note that the `2.1.0-SNAPSHOT` in the CI is also present on `main` - you might want to update that to `3.0.0-SNAPSHOT` there?
what's even the point of that step on CI? it just updates it locally but then never uses it...?

---

> i had understood that we should always depend on the snapshot version here, based on [this documentation](https://github.com/opensearch-project/opensearch-plugins/blob/main/BUILDING.md#developing-new-plugins-for-opensearch).

I think if you kick of a new plugin, the template against last released version would be what one is looking for: this is very likely what is deployed and could be tested against right away. 

> note that the 2.1.0-SNAPSHOT in the CI is also present on main - you might want to update that to 3.0.0-SNAPSHOT there?

Totally, I will take care of that, thank you

>  what's even the point of that step on CI? it just updates it locally but then never uses it...?

I don't know :D